fluentd/templates/fluent-config-template.conf

```diff
@@ -0,0 +1,14 @@
+{%- macro fluent_config_block(settings, offset=0) -%}
+{%- for setting in settings %}
+{{ ''|indent(offset, true) }}<{{ setting.directive }} {{ setting.get('directive_arg', '')}}>
+{% for attr in setting.attrs -%}
+{%- if 'nested_directives' in attr.keys() -%}
+{{ fluent_config_block(attr.nested_directives, offset + 2) }}
+{% else -%}
+{{ attr.keys()[0]|indent(offset + 2, true) }} {{ attr.values()[0] }}
+{% endif -%}
+{% endfor -%}
+{{ ''|indent(offset, true) }}</{{ setting['directive'] }}>
+{% endfor -%}
+{%- endmacro -%}
+{{ fluent_config_block(settings) }}
```

Comment:
This is a clever, but I'm okay with it. This config format mapping scares me a bit, although I don't particularly like the config format of fluentd (http://docs.fluentd.org/articles/config-file).